packages/go/analysis/ad/adcs.go

```diff
@@ -216,21 +216,46 @@ func PostEnrollOnBehalfOf(certTemplates []graph.Node, operation analysis.StatTra
 	}
 
 	operation.Operation.SubmitReader(func(ctx context.Context, tx graph.Transaction, outC chan<- analysis.CreatePostRelationshipJob) error {
-		return enrollOnBehalfOfVersionTwo(tx, versionTwoTemplates, certTemplates, outC)
+		if results, err := EnrollOnBehalfOfVersionTwo(tx, versionTwoTemplates, certTemplates); err != nil {
+			return err
+		} else {
+			for _, result := range results {
+				outC <- result
+			}
+
+			return nil
+		}
 	})
 
 	operation.Operation.SubmitReader(func(ctx context.Context, tx graph.Transaction, outC chan<- analysis.CreatePostRelationshipJob) error {
-		return enrollOnBehalfOfVersionOne(tx, versionOneTemplates, certTemplates, outC)
+		if results, err := EnrollOnBehalfOfVersionOne(tx, versionOneTemplates, certTemplates); err != nil {
+			return err
+		} else {
+			for _, result := range results {
+				outC <- result
+			}
+
+			return nil
+		}
 	})
 
 	operation.Operation.SubmitReader(func(ctx context.Context, tx graph.Transaction, outC chan<- analysis.CreatePostRelationshipJob) error {
-		return enrollOnBehalfOfSelfControl(tx, versionOneTemplates, outC)
+		if results, err := EnrollOnBehalfOfSelfControl(tx, versionOneTemplates); err != nil {
+			return err
+		} else {
+			for _, result := range results {
+				outC <- result
```

Comment:
It seems that the value of `channel.Submit` is checked in a lot of our existing post processing while writing to the channel. I'm not sure under what circumstances the context might expire while attempting to submit the value to the channel but I noticed this is submitting to the channel directly without that check. I don't have any actionable requests but it is something I noticed that is different from other post processing handlers so it might warrant some thought

Reply:
This was opened up recently in the AzureHound repo and looks like a similar situation:
https://github.com/BloodHoundAD/AzureHound/pull/61